compiler/src/iree/compiler/Codegen/Common/GPU/GPUDistributionPatterns.cpp

```diff
@@ -99,6 +188,208 @@ struct DistributeElementwise final : OpDistributionPattern<OpTy> {
+
+    replaceOpWithDistributedValues(rewriter, readOp, readVec);
+    return success();
+  }
+
+  VectorValue accessUnit(vector::TransferReadOp readOp,
+                         SmallVector<Value> &memoryIndices,
+                         SmallVector<int64_t> &accIndices,
+                         VectorValue accumulator, LayoutAttr vectorLayout,
+                         LayoutAttr memoryLayout,
+                         PatternRewriter &rewriter) const override {
+    auto unitType = VectorType::get({getLoadStoreWidth(memoryLayout)},
+                                    accumulator.getType().getElementType());
+    VectorValue load = rewriter.create<vector::LoadOp>(
+        readOp.getLoc(), unitType, readOp.getSource(), memoryIndices);
+    return rewriter.create<vector::InsertStridedSliceOp>(
+        readOp.getLoc(), load, accumulator, accIndices,
+        SmallVector<int64_t>{1});
+  }
+};
+
+struct DistributeTransferWriteLayoutAttr final
+    : DistributeXferLayoutAttr<vector::TransferWriteOp> {
+  using DistributeXferLayoutAttr::DistributeXferLayoutAttr;
+
+  LogicalResult matchAndRewrite(vector::TransferWriteOp writeOp,
+                                DistributionSignature &signature,
+                                PatternRewriter &rewriter) const override {
+    VectorValue vector = writeOp.getVector();
+    LayoutAttr vectorLayout = dyn_cast<LayoutAttr>(signature.operands[0]);
+    if (!vectorLayout) {
+      return failure();
+    }
+
+    // TODO: Return failure if we need masking.
+
+    Type elementType = writeOp.getSource().getType().getElementType();
+    auto vectorType = VectorType::get(
+        vectorLayout.getDistributedShape(vector.getType()), elementType);
+    Value zero = rewriter.create<arith::ConstantOp>(
+        writeOp.getLoc(), vectorType, rewriter.getZeroAttr(vectorType));
+    VectorValue acc = cast<VectorValue>(zero);
+
+    accessMemory(writeOp, acc, vectorLayout, rewriter);
+
+    rewriter.eraseOp(writeOp);
+    return success();
+  }
+
+  VectorValue accessUnit(vector::TransferWriteOp writeOp,
+                         SmallVector<Value> &memoryIndices,
+                         SmallVector<int64_t> &accIndices,
+                         VectorValue accumulator, LayoutAttr vectorLayout,
+                         LayoutAttr memoryLayout,
+                         PatternRewriter &rewriter) const override {
+    int width = getLoadStoreWidth(memoryLayout);
+
+    SmallVector<int64_t> strides(accIndices.size(), 1);
+    SmallVector<int64_t> shapes(accIndices.size(), 1);
+    shapes[shapes.size() - 1] = width;
+    Value result = rewriter.create<vector::ExtractStridedSliceOp>(
+        writeOp.getLoc(), getDistributed(rewriter, accumulator, vectorLayout),
+        accIndices, shapes, strides);
+    result = rewriter.create<vector::ExtractOp>(
+        writeOp.getLoc(), result,
+        SmallVector<int64_t>(accIndices.size() - 1, 0));
+    rewriter.create<vector::StoreOp>(writeOp.getLoc(), result,
```

Comment:
For here and above, if vector size = 1, will the vector.load/store lower to memref.load/store ?

Reply:
Yes, we would emit vector.load/vector.store and let further lowering get it to memref.load/store. I think that should be handled by llvm/further lowerings and we don't need to care about it here. I have tested this with width=1 and it makes no difference if I emit memref.load or vector.load.

Reply:
Great thanks for checking.